enhancements/network/primary-user-defined-networks-for-virtualization-workloads.md

```diff
@@ -0,0 +1,588 @@
+  automatically in CI) and by whom (e.g. perf team; name the responsible person and let them review
+  this enhancement)
+
+- Describe the possible failure modes of the API extensions.
+- Describe how a failure or behaviour of the extension will impact the overall cluster health
+  (e.g. which kube-controller-manager functionality will stop working), especially regarding
+  stability, availability, performance and security.
+- Describe which OCP teams are likely to be called upon in case of escalation with one of the failure modes
+  and add them as reviewers to this enhancement.
+
+## Support Procedures
+
+N/A
+
+## Alternatives
+
+### Binding mechanism
+
+We could adapt the existing (in-tree) bridge binding to bind to the UDN
+interface, and use it instead of passt. This would mimic the approach taken
+both for secondary layer2 and localnet networks (which use
+[persistent IPs](https://github.com/openshift/enhancements/pull/1456)) and
+also HyperShift (which relies on
+[point to point routing](https://github.com/openshift/enhancements/blob/master/enhancements/network/ovn-hypershift-live-migration.md#topology)).
+Doing so would solve the TCP connection reset issue we currently face on passt.
+
+If we implemented
+[this feature request](https://issues.redhat.com/browse/CNV-32573) in
+OVN-Kubernetes we have an extremely simple in-tree binding that would be
+responsible for layer2 connectivity: OVN-Kubernetes would provision DHCP flows
+for the VMs in the logical switches implementing the flat layer2 networks. To
+integrate with network policies pod / namespace selectors we would need to have
+a way for the pod to report the IPs effectively "owned" by the VM.
+
+There are some drawbacks to bridge binding though, namely:
+- would currently only work for IPv4. We would need to advertise routes either
+from OVN-Kubernetes, or from within the pod (which would require CAP_NET_ADMIN,
+which we do **not** want to reintroduce).
+- integrating with pod network ecosystem (service mesh, metrics, etc) difficult
+or impossible.
+
+Given these limitations we are currently focusing on passt. It could be an
+option assuming not having IPv6 support, nor integrating with service meshes or
+metrics is a possibility.
+
+**Note:** we could mitigate the first drawback (not having IPv6) support if we
+extended the DHCP OVN flow behavior to DHCPv6 and sent RAs from the GW routers for
+primary UDN networks. This would require more work in the SDN side of the
+integration. This would leave service mesh / ACS / metric integration as the
+drawback of this option.
+
+### VM interface IP address configuration
+
+We will also rely on the `IPAMClaim` CRD to provide the VM owner a way to
+specify the IP addresses for the UDN interface of their VMs. This way, we
+do not require to update the KubeVirt API, which is something we have faced
+extreme resistance in the past.
+
+Keep in mind that up to now the IPAMClaims would be created by KubeVirt
```

Comment:
I'm not sure about the details, but shouldn't this say "by OpenShift Virtualization directly"? Since it is created by IPAM extensions (?)

Reply:
Good catch; this was a left-over.